google-cloud-pubsub/src/main/java/com/google/cloud/pubsub/v1/MessageDispatcher.java

```diff
@@ -439,133 +400,27 @@ public void run() {
-              new PendingModifyAckDeadline(extensionSeconds);
-          for (AckHandler ackHandler : job.ackHandlers) {
-            pendingModAckDeadline.addAckId(ackHandler.ackId);
-          }
-          modifyAckDeadlinesToSend.add(pendingModAckDeadline);
-          renewJobs.add(job);
-        }
-        for (ExtensionJob job : renewJobs) {
-          outstandingAckHandlers.add(job);
-        }
-        if (!outstandingAckHandlers.isEmpty()) {
-          nextScheduleExpiration = outstandingAckHandlers.peek().expiration;
-        }
-      }
-
-      processOutstandingAckOperations(modifyAckDeadlinesToSend);
-
-      if (nextScheduleExpiration != null) {
-        logger.log(
-            Level.FINER,
-            "Scheduling based on outstanding, at time: {0}, next scheduled time: {1}",
-            new Object[] {now, nextScheduleExpiration});
-        setupNextAckDeadlineExtensionAlarm(nextScheduleExpiration);
+  @VisibleForTesting void extendDeadlines() {
+    List<String> acksToSend = Collections.<String>emptyList();
+    PendingModifyAckDeadline modack = new PendingModifyAckDeadline(getMessageDeadlineSeconds());
+    Instant now = now();
+
+    Iterator<Map.Entry<String, Instant>> it = pendingMessages.entrySet().iterator();
+    while (it.hasNext()) {
+      Map.Entry<String, Instant> entry = it.next();
+      if (entry.getValue().isBefore(now)) {
+        it.remove();
+      } else {
+        modack.ackIds.add(entry.getKey());
       }
     }
-  }
 
-  private void setupNextAckDeadlineExtensionAlarm(Instant expiration) {
-    Instant possibleNextAlarmTime = expiration.minus(ackExpirationPadding);
-    alarmsLock.lock();
-    try {
-      if (nextAckDeadlineExtensionAlarmTime.isAfter(possibleNextAlarmTime)) {
-        logger.log(
-            Level.FINER,
-            "Scheduling next alarm time: {0}, previous alarm time: {1}",
-            new Object[] {possibleNextAlarmTime, nextAckDeadlineExtensionAlarmTime});
-        if (ackDeadlineExtensionAlarm != null) {
-          logger.log(Level.FINER, "Canceling previous alarm");
-          ackDeadlineExtensionAlarm.cancel(false);
-        }
-
-        nextAckDeadlineExtensionAlarmTime = possibleNextAlarmTime;
-
-        ackDeadlineExtensionAlarm =
-            systemExecutor.schedule(
-                new AckDeadlineAlarm(),
-                nextAckDeadlineExtensionAlarmTime.toEpochMilli() - clock.millisTime(),
-                TimeUnit.MILLISECONDS);
-      }
-
-    } finally {
-      alarmsLock.unlock();
-    }
-  }
-
-  private void processOutstandingAckOperations() {
-    processOutstandingAckOperations(Collections.<PendingModifyAckDeadline>emptyList());
+    ackProcessor.sendAckOperations(acksToSend, Collections.singletonList(modack));
   }
 
-  private void processOutstandingAckOperations(
-      List<PendingModifyAckDeadline> ackDeadlineExtensions) {
-    List<PendingModifyAckDeadline> modifyAckDeadlinesToSend =
-        Lists.newArrayList(ackDeadlineExtensions);
+  @VisibleForTesting
```

Comment:
Please use `@InternalApi` (from api-common) instead of `@VisibleForTesting` (from Guava).

Reply:
Done